services/httpd/pprof.go

```diff
@@ -0,0 +1,151 @@
+package httpd
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"fmt"
+	"net/http"
+	httppprof "net/http/pprof"
+	"runtime/pprof"
+	"strconv"
+	"time"
+)
+
+// handleProfiles determines which profile to return to the requester.
+func handleProfiles(w http.ResponseWriter, r *http.Request) {
+	switch r.URL.Path {
+	case "/debug/pprof/cmdline":
+		httppprof.Cmdline(w, r)
+	case "/debug/pprof/profile":
+		httppprof.Profile(w, r)
+	case "/debug/pprof/symbol":
+		httppprof.Symbol(w, r)
+	case "/debug/pprof/all":
+		archiveProfiles(w, r)
+	default:
+		httppprof.Index(w, r)
+	}
+}
+
+// prof describes a profile name and a debug value, or in the case of a CPU
+// profile, the number of seconds to collect the profile for.
+type prof struct {
+	Name  string
+	Debug int64
+}
+
+// archiveProfiles collects the following profiles:
+//	- goroutine profile
+//	- heap profile
+//	- blocking profile
+//	- (optionally) CPU profile
+//
+// All profiles are added to a tar archive and then compressed, before being
+// returned to the requester as an archive file.  To optionally include a CPU
+// profile, the requester should provide a `cpu` query parameter, and can also
+// provide a `seconds` parameter to specify a non-default profile collection
+// time. The default CPU profile collection time is 30 seconds.
+//
+// Example request including CPU profile:
+//
+//	http://localhost:8086/debug/pprof/all?cpu=true&seconds=45
+//
+// The value after the `cpu` query parameter is not actually important, as long
+// as there is something there.
+//
+func archiveProfiles(w http.ResponseWriter, r *http.Request) {
+	var all = []*prof{
+		{Name: "goroutine", Debug: 1},
```

Comment:
The `goroutine` profile accepts debug=2, but I think we're always more interested in debug=1 output, since it combines goroutines with identical stack traces. Maybe add a comment noting that we deliberately aren't supporting debug=2?